src/utils/fetchUtils.ts

```diff
@@ -55,31 +55,32 @@ export class FetchResponse extends Response {
     return headers
   }
 
-  constructor(body?: BodyInit | null, init: FetchResponseInit = {}) {
+  constructor(body?: BodyInit | null, init: FetchResponseInit = {}) { 
+    super()
     const status = init.status ?? 200
     const safeStatus = FetchResponse.isConfigurableStatusCode(status)
       ? status
       : 200
     const finalBody = FetchResponse.isResponseWithBody(status) ? body : null
-
-    super(finalBody, {
+    // construct the response manually so it will use the Response proxy for raw headers
+    const response = Reflect.construct(Response, [finalBody, {
```

Comment:
My only concern with this is that it messes up the prototype chain. We can no longer do things like `x instanceof FetchResponse` because the class constructor returns an arbitrary object (a plain Fetch API Response instance).

Reply:
agree